### PR TITLE
selinux-tests.py: add distribution specific test cases

### PR DESCRIPTION
**Pull request by nasastry**
Add distribution specific test cases.
With the support of yaml, can run distribution and upstream test cases.

Signed-off-by: Nageswara R Sastry <rnsastry@linux.ibm.com>

### CODE REVIEW

**Comment by nasastry** [comment]
```
RHEL:
JOB ID     : 9d2e3e0958848846a2c982b1febefe37e7426ef2
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-09-26T15.30-9d2e3e0/job.log
 (1/2) selinux-tests.py:SELinux.test;run-run_type-upstream-0113: CANCEL: SELinux tests not supported on SuSE (0.51 s)
 (2/2) selinux-tests.py:SELinux.test;run-run_type-distro-9cc6: CANCEL: SELinux tests not supported on SuSE (0.53 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 2
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-09-26T15.30-9d2e3e0/results.html
JOB TIME   : 19.78 s


SLES:
JOB ID     : 9d2e3e0958848846a2c982b1febefe37e7426ef2
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-09-26T15.30-9d2e3e0/job.log
 (1/2) selinux-tests.py:SELinux.test;run-run_type-upstream-0113: CANCEL: SELinux tests not supported on SuSE (0.51 s)
 (2/2) selinux-tests.py:SELinux.test;run-run_type-distro-9cc6: CANCEL: SELinux tests not supported on SuSE (0.53 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 2
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-09-26T15.30-9d2e3e0/results.html
JOB TIME   : 19.78 s

```

File: security/selinux-tests.py [hunk 1]
```diff
@@ -34,44 +34,54 @@ def setUp(self):
...
+                        % detected_distro.name)
         for package in deps:
             if not smm.check_installed(package) and not smm.install(package):
                 self.cancel('%s is needed for the test to be run' % package)
-        url = "https://github.com/SELinuxProject/selinux-testsuite/archive/master.zip"
-        tarball = self.fetch_asset(url, expire='7d')
-        archive.extract(tarball, self.workdir)
-        self.sourcedir = os.path.join(self.workdir, 'selinux-testsuite-master')
-        os.chdir(self.sourcedir)
+        run_type = self.params.get('type', default='upstream')
```
Comment (Naresh-ibm) [inline security/selinux-tests.py:55]: @nasastry change default='distro'

File: security/selinux-tests.py [hunk 2]
```diff
@@ -34,44 +34,54 @@ def setUp(self):
...
+        if run_type == "upstream":
+            default_url = ("https://github.com/SELinuxProject/"
+                           "selinux-testsuite/archive/master.zip")
+            url = self.params.get('url', default=default_url)
```
Comment (Naresh-ibm) [inline security/selinux-tests.py:59]: remove def_url variable and change default=None

**Review by Naresh-ibm — CHANGES_REQUESTED** [review]

**Review by abdhaleegit — APPROVED** [review]
LGTM

**Review by Naresh-ibm — APPROVED** [review]
Thanks @abdhaleegit @nasastry 
LGTM

**Pull request merged**